roles/cnf_cert/tasks/tests.yml

```diff
@@ -11,37 +23,94 @@
   environment: "{{ tnf_env_vars | combine({'TNF_PARTNER_REPO': tnf_partner_repo, 'TNF_IMAGE': tnf_image, 'SUPPORT_IMAGE': support_image}) }}"
   ansible.builtin.shell: |
     set -x
 
-    ./run-tnf-container.sh -k {{ kubeconfig_path }} \
-    -t {{ tnf_dir }}/{{ test_network_function_project_name }} \
-    -o {{ tnf_dir }}/{{ test_network_function_project_name }} \
+    ./run-tnf-container.sh -k {{ tnf_dir }}/config_files/kubeconfig \
+    -t {{ tnf_dir }}/config_files \
+    -o {{ tnf_dir }}/result_files \
     {% if tnf_labels | length > 0 %}-l '{{ tnf_labels }}' \{% endif %}
     {% if tnf_dockercfg_path is defined %}-c {{ tnf_dockercfg_path }} \{% endif %}
-    -i {{ tnf_image }} &> {{ test_network_function_project_name }}/dci-tnf-execution.log
+    -i {{ tnf_image }} &> result_files/dci-tnf-execution.log
   args:
     chdir: "{{ tnf_dir }}"
   ignore_errors: "{{ tnf_exec_ignore_errors | bool }}"
 
+# For v5.1.1 in advance, keep using some environment variables to
+# keep the definition of tnf_env_vars, but would be deprecated at some point.
+- name: Cpmmon tasks for CNF Cert Suite execution (tnf >= v5.1.1 and HEAD)
+  when: test_network_function_version is version("v5.1.1", ">=") or test_network_function_version == "HEAD"
+  block:
+    - name: Run the CNF Test Suite (tnf >= v5.1.1 and HEAD)
+      vars:
+        tnf_partner_repo: "{{ (dci_local_registry | length) | ternary(dci_local_registry, 'quay.io') }}/testnetworkfunction"
+        support_image: "debug-partner:{{ support_image_version }}"
+        container_command: |
+          ./cnf-certification-test/certsuite run \
+          {% if tnf_labels | length > 0 %}
+          --label-filter='{{ tnf_labels }}' \
+          {% endif %}
+          {% if tnf_dockercfg_path is defined %}
+          --preflight-dockerconfig=/usr/tnf/config/config.json \
+          --allow-preflight-insecure=${TNF_ALLOW_PREFLIGHT_INSECURE:-false} \
+          {% endif %}
+          --output-dir=/usr/tnf/results \
+          --offline-db=/usr/offline-db \
+          --config-file=/usr/tnf/config/tnf_config.yml \
+          --kubeconfig=/usr/tnf/config/kubeconfig \
+          --tnf-image-repository={{ tnf_partner_repo }} \
+          --tnf-debug-image={{ support_image }} \
+          --create-xml-junit-file=${TNF_ENABLE_XML_CREATION:-true} \
+          --non-intrusive=${TNF_NON_INTRUSIVE_ONLY:-false} \
+          --log-level=${TNF_LOG_LEVEL:-debug} \
+          --include-web-files=true
+      environment: "{{ tnf_env_vars }}"
+      ansible.builtin.shell: |
+        set -x
+
+        podman run \
+        --rm \
+        --network host \
+        -v {{ tnf_dir }}/config_files:/usr/tnf/config:Z \
+        -v {{ tnf_dir }}/result_files:/usr/tnf/results:Z \
+        {{ tnf_image }} \
+        {{ container_command }}
+      register: tnf_output
+      no_log: true
+
+    - name: Save output in a file
+      ansible.builtin.copy:
```

Comment:
If you are using the `no_log` option above, you may want to use it here as well.

Reply:
Not really required because the output cannot be seen in this task but I've done it just in case, thanks!